pkg/controller/node.go

```diff
@@ -83,7 +90,7 @@ func (c *Controller) moveMasterPodsOffNode(node *v1.Node) {
 	podList, err := c.KubeClient.Pods(c.opConfig.WatchedNamespace).List(opts)
 	if err != nil {
 		c.logger.Errorf("could not fetch list of the pods: %v", err)
-		return
+		return err
```

Comment:
If I understand correctly, `attemptToMoveMasterPodsOffNode` will be retried only if returned an error, i.e. only if we're on this line. The rest of the errors, that could happen (including switchover itself), are sort of hidden in `range masterPods` loop. Is that intended behaviour?

Reply:
> `attemptToMoveMasterPodsOffNode` will be retried only if returned an error

yes.

> The rest of the errors, that could happen (including switchover itself), are sort of hidden in `range masterPods` loop

this PR also introduces the following change 
> 	if leftPods := totalPods - movedPods; leftPods > 0 {
		       return fmt.Errorf("could not move master %d/%d pods from the %q node",
			leftPods, totalPods, nodeName)
	}

so whatever goes wrong is supposed to be caught here